Update nodejs versions

edmorley opened the pull request:
Since amethyst only has one nodejs version installed, I've left it at the most recent LTS release (the v6 series). For garnet I've used latest LTS (v6) and latest stable (v8).

~For sugilite, I've added the latest stable (v8), however to keep the image size comparable, I've also dropped the long ago EOLed v0.12 series. We could probably also drop the v4 and v5 series, but it would be good to know some usage stats first to avoid making lots of jobs have to download at runtime.~
(I've dropped the sugilite part of this PR given it's about to be removed in #498)

Version numbers taken from:
https://nodejs.org/en/download/releases/

Lyoness commented:
>It'd be really nice to keep 0.12 and 0.10 preinstalled - is there a cost to doing that?

@ljharb we like to keep only the most-used and let the user download special versions if they need.  

But @edmorley 's comment: 
>would be good to know some usage stats first. 

I can try and query the most frequent node versions specified in the .travis.yml configs and adjust accordingly for the next trusty update  (as well our just-started-development on xenial) 

I am not sure if we can catch `before_install` or `script` block run-time node installs, but this gives us a better understanding regardless.

Lyoness commented:
@edmorley @ljharb here is our data on node versions: 

![screenshot from 2017-08-25 15 40 23](https://user-images.githubusercontent.com/1622229/29729844-d76b2ae2-89ab-11e7-9f90-f938b66d5490.png)

0.10 and 0.12 are a small percentage.

I've discussed with @BanzaiMan Node.js installation is pretty quick, so adding it to a script at run-time might be a better solution. 

re: disk space, a quick query

```
[~/.nvm/versions/node]$ du -hs *
130M    v6.6.0
 51M    v6.9.1
 62M    v7.7.1
 59M    v8.1.3
 62M    v8.4.0
```
makes me feel like we can keep .12 to ci-sugilite.

Lyoness commented:
[sugilite](https://travis-ci.org/travis-infrastructure/packer-build/builds/268463460) :white_check_mark: 
[garnet](https://travis-ci.org/travis-infrastructure/packer-build/builds/268463433) :white_check_mark: 
[amethyst](https://travis-ci.org/travis-infrastructure/packer-build/builds/268463409) :white_check_mark: 
[connie](https://travis-ci.org/travis-infrastructure/packer-build/builds/268463416) :white_check_mark: 

[steveonnie](https://travis-ci.org/travis-infrastructure/packer-build/builds/268463427) <- our new minimal image for xenial :white_check_mark:

Lyoness commented:
The new dev images have node `v8.4.0`
https://travis-ci.org/Lyoness/travis_production_test/jobs/268501156#L475

(current stable defaults to `v7.4.0`)

Lyoness commented:
we can merge as-is or we can add `0.12` back in.

edmorley commented:
I've dropped the sugilite part of this PR given it's about to be removed in #498, which avoids the need to make a decision on 0.12. I've also rebased on master and mirrored the amethyst/garnet changes to opal/sardonyx now that those images exist.

Test builds (I lumped this in with the Docker PR testing):
* [amethyst](https://travis-ci.org/travis-infrastructure/packer-build/builds/269816802)
* [garnet](https://travis-ci.org/travis-infrastructure/packer-build/builds/269816799)

I've not included the opal/sardonyx builds here, since they are failing but not due to this PR (see travis-ci/packer-templates#501).

ljharb commented:
For what it's worth, my 0.6 builds have recently stopped working because 0.6 takes too long to compile - it'd be *really nice* if it could remain preinstalled (https://travis-ci.org/ljharb/node-resolve/jobs/271020794 used to work, and doesn't now)

edmorley commented:
@ljharb 0.6 isn't pre-installed on Trusty at the moment anyway? Also looking at that log, the job wasn't terminated due to taking too long, but due to exceeding the 4MB log allowance. If you manage to suppress the compile warnings spew it should probably succeed :-)

ljharb commented:
@edmorley true; but I'm pretty sure it didn't used to need compiling at all. I'm not sure how to suppress the compiler warnings since travis, not my build, is running the `nvm install` command - but as recently as a week or two ago, 0.6 didn't cause these compiler warnings to happen :-/

edmorley commented:
@ljharb that's strange - however I'd open an issue [here](https://github.com/travis-ci/travis-ci) or else contact Travis support via email, since it's unrelated to this not yet merged PR.
